todo_app/data/session_items.py

```diff
@@ -1,69 +1,92 @@
+            { 'id': 2, 'status': 'Not Started', 'title': 'Allow new items to be added' }
+        ]
 
-def get_items():
-    """
-    Fetches all saved items from the session.
+    def get_items(self):
+        """
+        Fetches all saved items from the session.
 
-    Returns:
-        list: The list of saved items.
-    """
-    return session.get('items', _DEFAULT_ITEMS.copy())
+        Returns:
+            list: The list of saved items.
+        """
+        return session.get('items', self._DEFAULT_ITEMS.copy())
 
 
-def get_item(id):
-    """
-    Fetches the saved item with the specified ID.
+    def get_item(self, id):
+        """
+        Fetches the saved item with the specified ID.
 
-    Args:
-        id: The ID of the item.
+        Args:
+            id: The ID of the item.
 
-    Returns:
-        item: The saved item, or None if no items match the specified ID.
-    """
-    items = get_items()
-    return next((item for item in items if item['id'] == int(id)), None)
+        Returns:
+            item: The saved item, or None if no items match the specified ID.
+        """
+        items = self.get_items()
+        return next((item for item in items if item['id'] == int(id)), None)
 
 
-def add_item(title):
-    """
-    Adds a new item with the specified title to the session.
+    def add_item(self, title):
+        """
+        Adds a new item with the specified title to the session.
 
-    Args:
-        title: The title of the item.
+        Args:
+            title: The title of the item.
 
-    Returns:
-        item: The saved item.
-    """
-    items = get_items()
+        Returns:
+            item: The saved item.
+        """
+        items = self.get_items()
 
-    # Determine the ID for the item based on that of the previously added item
-    id = items[-1]['id'] + 1 if items else 0
+        # Determine the ID for the item based on that of the previously added item
+        id = items[-1]['id'] + 1 if items else 0
 
-    item = { 'id': id, 'title': title, 'status': 'Not Started' }
+        item = { 'id': id, 'title': title, 'status': 'Not Started' }
 
-    # Add the item to the list
-    items.append(item)
-    session['items'] = items
+        # Add the item to the list
+        items.append(item)
+        session['items'] = items
 
-    return item
+        return item
+    
+    def delete_item(self, id):
+        """
+        Deletes an item with the specified id to the session.
 
+        Args:
+            id: The id of the item.
 
-def save_item(item):
-    """
-    Updates an existing item in the session. If no existing item matches the ID of the specified item, nothing is saved.
+        """
+        id = int(id)
+        items = self.get_items()
+        deleted = False
+        items.remove(items[id-1])
+        # Drop ID of following items by 1
+        for item in items:
+            if item['id'] > id:
+                item['id'] = int(item['id']) - 1
```

Comment:
This works, but changing IDs after they're set is always a risk, as if anything goes wrong you could find it very hard to work out what it was previously (and if we later have any ties from id to other fields, e.g. foreign keys in a database, using their id as a reference in external systems, or for audit purposes, we could easily break those links)

A better pattern is probably just finding the item & removing it without modifying ids:
```suggestion
        items = self.get_items()
        for item in items:
            if item['id'] == id:
                items.remove(item)
```

Or alternatively, not necessarily a recommendation but worth seeing for comparison, a list-comprehension approach:
```python
        items = self.get_items()
        items = [item for item in items if item['id'] != id]
```